### PR TITLE
parts: unmarshal plugin properties

### PR DESCRIPTION
**Pull request by cmatsuoka**
Plugin properties are specified along part properties and prefixed by
the plugin name. Validate and unmarshal properties and assign them
to the corresponding part.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: craft_parts/lifecycle_manager.py [hunk 1]
```diff
@@ -103,7 +104,44 @@ def plan(self, target_step: Step, part_names: Sequence[str] = None) -> List[Acti
...
+
+def _build_part(name: str, spec: Dict[str, Any], project_dirs: ProjectDirs) -> Part:
+    if not isinstance(spec, dict):
+        raise errors.PartSpecificationError(
+            part_name=name, message="part definition is malformed"
+        )
+
+    plugin_name = spec.get("plugin", "")
+
+    fallback_plugin_name = not plugin_name
```
Comment (sergiusens) [inline craft_parts/lifecycle_manager.py:119]:
```suggestion
    part_name_as_plugin_name = not plugin_name
```

**Review by sergiusens — APPROVED** [review]
approve, with the caveat of variable change (or code comments) as I only understood why this was needed at the end of the story

File: craft_parts/lifecycle_manager.py (hunk 1 above)
Comment (cmatsuoka) [reply craft_parts/lifecycle_manager.py]: Renamed variable, and added a comment as well.

**Review by facundobatista — APPROVED** [review]
Nice! Just check one inline detail.

**Pull request merged**